mwclient/client.py

```diff
@@ -1257,21 +1457,44 @@ def blocks(self, start=None, end=None, dir='older', ids=None, users=None, limit=
         return listing.List(self, 'blocks', 'bk', max_items=max_items,
                             api_chunk_size=api_chunk_size, **kwargs)
 
-    def deletedrevisions(self, start=None, end=None, dir='older', namespace=None,
-                         limit=None, prop='user|comment', max_items=None,
-                         api_chunk_size=None):
+    def deletedrevisions(
+        self,
+        start: Optional[str] = None,
+        end: Optional[str] = None,
+        dir: str = 'older',
+        namespace: Optional[int] = None,
```

Comment:
should this be `Namespace`?

Reply:
According to the MediaWiki documentation, the only allowed values for this are integers (https://www.mediawiki.org/wiki/API:Deletedrevs#query+deletedrevs:drnamespace). Most other APIs also allow `*` (all namespaces). While we could technically allow `str`, since the function works fine with strings, using int enforces correct usage (i.e., no `*`).